src/main/java/seedu/duke/task/Deadline.java

```diff
@@ -0,0 +1,57 @@
+package seedu.duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Used when user wants to create a {@link Task} that has a end date but no start date.
+ */
+public class Deadline extends Task {
+    private final String taskType = "D";
+
+    /**
+     * Creates a deadline.
+     * @param taskName for task name
+     * @param date for end date
+     */
+    public Deadline(String taskName, LocalDateTime date) {
+        super(taskName, false, date);
+    }
+
+    /**
+     * Used to help adjust the done status.
+     * @param oldDeadline to extract task name and date
+     * @param done for specified boolean
+     */
+    Deadline(Deadline oldDeadline, boolean done) {
+        super(oldDeadline.getTaskName(), done, oldDeadline.getDate());
+    }
+
+    public Deadline(String taskName, boolean doneStatus, LocalDateTime date) {
```

Comment:
Perhaps consider renaming `doneStatus` to a more boolean sounding one like `isDone`. I think it's fine, but there's some room for ambiguity.

Reply:
Ah that is true, I have adjusted _doneStatus_ to _isDone_, I think it is clearer - thank you!